### DISCUSSION TITLE
How to get graph level embedding ?

### DISCUSSION THREAD

**Discussion by dxyzx0** [pyg-team/pytorch_geometric#7472] [Q&A]
1. While node-level and edge-level embeddings are common in graph theory, I'm curious about the best practices for obtaining graph-level embeddings. One approach that comes to mind is pooling all node features into a single vector, but I'm unsure if this is the most effective method. Are there established techniques for generating graph-level embeddings?
2. I've been trying to find research papers specifically focused on graph-level embeddings. While I've found some that discuss graph similarities, my interest lies in obtaining a single vector of embeddings for an individual graph. Could you recommend any relevant literature in this area?

Thanks for helping!

**Comment by wsad1** [comment]
Pyg has several pooling [operators](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#pooling-layers). You can find papers linked in the docs of the pooling operators. Also there are several examples of pooling operators [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/) like `proteins_mincut_pool.py`, `proteins_topk_pool.py` etc.

**Reply by dxyzx0** [reply]
Thanks a lot. I have checked this page, however my question is whether it's appropriate to just add a pooling layer to the end of model to get a graph level embedding. Maybe the answer is in the paper and I will try to find it out.

**Reply by rusty1s** [reply]
Yes, this is the most common method to pool node-level features to graph-level ones.

**Reply by dxyzx0** [reply]
Thanks, I get it.

I'm trying to perform a task of learning feature embedding for my graphs.

Is using an autoencoder the most commonly used approach for this purpose?

**Reply by rusty1s** [reply]
There exists a wide range of literature on unsupervised/self-supervised learning on graphs. I would assume autoencoder/link prediction is the most common one, but there exists others as well, such as Deep Graph Infomax.
